hypothesis-python/tests/cover/test_baseexception.py

```diff
@@ -0,0 +1,61 @@
+import pytest
+
+from hypothesis import assume, given, infer, reject, settings
+from hypothesis.errors import InvalidArgument, Unsatisfiable
+from hypothesis.strategies import integers, composite
+from tests.common.utils import fails_with
+
+
+def test_keyboardinterrupt_no_rerun():
+    runs = 0
+    interrupt = 3
+    @given(integers())
+    def test_raise_keyboardinterrupt(x):
+        nonlocal runs
```

Comment:
Unfortunately the `nonlocal` statement is not supported in Python 2, and we have [fifty days left](https://pythonclock.org/) before we drop it :grimacing: 

Our usual idiom would be `runs = [0]`, `runs[0] += 1`; `if runs[0] == interrupt`.  Sorry!